api/app/models/concerns/filterable.rb

```diff
@@ -3,14 +3,63 @@ module Filterable
 
```

Comment:
Look at my extended commit message for my thinking on this. I like where you're going with the filtering—getting the IDs from the active record query and passing them to elastic search if necessary—but having that logic spread out across all our models is going to get kind of crazy. I've refactored the filterable concern so that all filtering (and pagination) logic is in one place. The only thing that stays on the models are the named scopes, which seems correct. This is still a little complex, and we may want to tweak it, but at least it's now DRY.